### ISSUE TITLE
Interdependency between this and monorepo

### ISSUE THREAD

**Issue by L-as** [o1-labs/snarkyjs-bindings#51]
MinaProtocol/mina depends on this, but this also depends on the monorepo.
Very little is gained from this setup AFAICT compared to having it be part of the monorepo properly rather than as a submodule.
Alternatively you could break the cycle and remove the submodule from the monorepo.

**Comment by L-as** [comment]
I guess the point is that SnarkyJS can have this as a submodule and not the whole Mina repo, but is the code here really used? I assume it's essentially only the compiled result that is used, which shouldn't be part of this repo anyway.

**Comment by mitschabaude** [comment]
Hey @L-as, this repo has a lot of TS code that's used in snarkyjs.

Regarding the Rust/OCaml deps in here, as you said it greatly speed up snarkyjs development to not have to change the Mina repo / be tied to a Mina branch every time you change snarkyjs. The entangling of Mina and snarkyjs development was very bad for velocity. The future direction for this repo will likely be to move _more_ stuff (like snarky-ml, pickels etc) into direct dependencies of snarkjys, instead of being an "inverted" dependency via the Mina repo, so that snarkyjs can be developed largely independently of Mina

**Comment by mitschabaude** [comment]
The compiled result has to be part of the repo to enable developers to work on snarkyjs without having to build everything from source

**Comment by L-as** [comment]
Yes but that compiled result could be part of the _snarkyjs_ repo.

**Comment by mitschabaude** [comment]
> Yes but that compiled result could be part of the _snarkyjs_ repo.

That's true :D I guess it felt natural to put it next to the source code. Usually a change of the compiled artifacts will correspond to a change of the source code, so you get both of those changes in the same PR instead of split between repos
